### PR TITLE
GHA to require backend-review-group approval on PRs

### PR DESCRIPTION
**Pull request by rjohnson2011**
## Summary

- Introduces a GHA to require backend-review-group approval before PRs can be merged on vets-api
- Exceptions made for Identity, Mobile, Lighthouse teams

## Related issue(s)

- [Zenhub Ticket](https://app.zenhub.com/workspaces/platform-product-team-633af4074573d06c3cda142a/issues/gh/department-of-veterans-affairs/va.gov-team/88204)

## Testing Done
- Tested with adding identity / mobile / lighthouse reviewers and the GHA steps all pass.

### CODE REVIEW

File: .github/workflows/be_review_prs.yml
Comment (rmtolmach) [inline .github/workflows/be_review_prs.yml:1]: Can we change this name? I'd like to make it abundantly clear why it's failing so there's minimal confusion 😄 Maybe "Require backend-review-group approval" or "Backend-review-group approval required"?
image

File: .github/workflows/be_review_prs.yml
Comment (LindseySaari) [inline .github/workflows/be_review_prs.yml:52]: @rjohnson2011 Can we have this post a comment on the PR too? An example of where we create a PR comment [is here](https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/blob/main/.github/workflows/force-separate-prod-vs-lower-env-prs.yaml#L44-L49)

**Review by LindseySaari — APPROVED** [review]

**Review by kpethtel — APPROVED** [review]
Looks good as far as the mobile team is concerned.

**Review by LindseySaari — DISMISSED** [review]

**Comment by rmtolmach** [comment]
Two downsides: 
1. While on support, if you're looking at the list of open PRs, they will all show up with a ❌ for CI. So we can't use the green check to determine which PRs are ready to review or not.
2. It doesn't add `backend-review-group` as reviewers, so the PRs won't show up in our list of PRs needing reviewed. I could see this resulting in more support requests. I wonder if adding "Please add backend-review-group as reviewers (if they're not already listed)." would be helpful. I know sometimes we'll already be listed, but sometimes not.

I know we've talked about # 1 and we're gonna live with it, but @rjohnson2011, can you add some more text to the automated comment to address # 2?

**Review by rmtolmach — APPROVED** [review]

**Comment by rmtolmach** [comment]
After I approved, some CI checks ran again
image

**Comment by rmtolmach** [comment]
And after I approved, we still got the message about requiring a backend review and merging is blocked 😕 
image

File: .github/workflows/be_review_prs.yml
Comment (rmtolmach) [inline .github/workflows/be_review_prs.yml:29]: Can you test that these are excluded?

**Comment by rjohnson2011** [comment]
> After I approved, some CI checks ran again image

Yeah I noticed this behavior too, working on a fix now.

File: .github/workflows/be_review_prs.yml
Comment (rjohnson2011) [reply .github/workflows/be_review_prs.yml:29]: Yes, I did earlier and the CI workflow passed. I'll double check again since some code was modified.

**Review by stevenjcumming — APPROVED** [review]

**Review by stevenjcumming — APPROVED** [review]

**Review by stevenjcumming — APPROVED** [review]
approved these changes on behalf of https://github.com/orgs/department-of-veterans-affairs/teams/backend-review-group

**Review by stevenjcumming — DISMISSED** [review]
on behalf of department-of-veterans-affairs/backend-review-group

**Review by stevenjcumming — APPROVED** [review]
on behalf of department-of-veterans-affairs/backend-review-group

**Review by stevenjcumming — APPROVED** [review]

**Review by rmtolmach — APPROVED** [review]

**Review by kpethtel — APPROVED** [review]
Good afaict.

**Review by LindseySaari — APPROVED** [review]

**Pull request merged**
